Dockerfile

```diff
@@ -7,18 +7,17 @@ WORKDIR /stage
 RUN conda create -n runenv --yes python=3.5
 ENV PYTHONHASHSEED 2157
 
-# Copying and installing required python packages.
-COPY requirements.txt requirements.txt
-# requirements.txt uses the CPU version of tensorflow, for Travis CI; we want the GPU version, so
-# we use sed to change it before installing the requirements.
-RUN sed -ie 's/^tensorflow/tensorflow-gpu/g' requirements.txt
-RUN /bin/bash -c "source activate runenv && pip install -r requirements.txt"
-RUN /bin/bash -c "source activate runenv && python -m nltk.downloader punkt"
-RUN /bin/bash -c "source activate runenv && python -m spacy.en.download all"
-
-# Copying the source code.
+# Copying the source code and pip requirements.
 COPY scripts/ scripts/
 COPY deep_qa/ deep_qa/
+COPY requirements.txt requirements.txt
```

Comment:
This line should be earlier in the dockerfile than `COPY` lines for the scripts and the code.  This is because of how docker images are built on top of each other.  This line is the most expensive image to build, because it's grabbing and installing a bunch of packages, and the requirements change less frequently than the code does.  So you don't want to have to rebuild the requirements every time you change your code.  I would move the current lines 11 and 12 down below line 20, to maintain the nice build behavior we currently have.